xorgauth/accounts/views.py

```diff
@@ -51,6 +58,56 @@ def user_xorg_email(self):
         return user.main_email if user.is_x_alumni() else None
 
 
+class SyncAxData(View):
+    def post(self, request, *args, **kwargs):
+        """Receive some data for syncing"""
+        # Do not do anything if no secret has been defined
+        if not settings.AX_SYNC_SECRET_CRYPT:
+            raise Http404("AX sync has not been configured")
+
+        try:
+            data = json.loads(request.body.decode('ascii'))
+        except ValueError:
+            return HttpResponseBadRequest("Unable to load request")
+
+        # data['secret'] is a password for authenticating the data provider
+        digest = crypt.crypt(data.get('secret'), settings.AX_SYNC_SECRET_CRYPT)
+        if not hmac.compare_digest(digest, settings.AX_SYNC_SECRET_CRYPT):
+            return HttpResponseForbidden("Unauthenticated")
+
+        # Mapping from User model to value in JSON request
+        field_mapping = (
+            ('alumnforce_id', 'af_id'),
+            ('ax_contributor', 'ax_contributor'),
+            ('axjr_subscriber', 'axjr_subscribed'),
+            ('ax_last_synced', 'last_updated'),
+        )
+        try:
+            for account_data in data['data']:
+                try:
+                    user = User.objects.get(hrid=account_data['xorg_id'])
+                except User.DoesNotExist:
+                    # Silently skip users that do not exist
+                    # If they are created, they will be synced the next time a
+                    # synchronisation request is sent to xorgauth.
+                    continue
+
+                # Change the account and commit the object only if there were some changes
+                changed = False
+                for model_field, json_field in field_mapping:
+                    if json_field not in account_data:
+                        continue
+                    new_value = account_data[json_field]
+                    if getattr(user, model_field) != new_value:
+                        setattr(user, model_field, new_value)
+                        changed = True
```

Comment:
Would it be relevant to introduce some whitelist of which fields the AX Data Sync is allowed to update, just in case?

Reply:
There already is such a whitelist, in variable `field_mapping` just above ;)

Reply:
Oh right, my bad.